tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/LanguageConstructDebugTest.java

```diff
@@ -0,0 +1,125 @@
+import org.testng.annotations.Test;
+
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Comparator;
+
+/**
+ * Test class for language construct related debug scenarios.
+ */
+public class LanguageConstructDebugTest extends DebugAdapterBaseTestCase {
+
+    Comparator<Variable> compareByName = Comparator.comparing(Variable::getName);
+
+    @BeforeClass
+    public void setup() {
+        testProjectName = "breakpoint-tests";
+        testModuleName = "languageConstruct";
+        testModuleFileName = "mainLangConstruct.bal";
+        testProjectPath = Paths.get(testProjectBaseDir.toString(), testProjectName).toString();
+        testEntryFilePath = Paths.get(testProjectPath, "src", testModuleName, testModuleFileName).toString();
+    }
+
+    @Test
+    public void testLanguageConstructDebugScenarios() throws BallerinaTestException {
+        addBreakPoint(new BallerinaTestDebugPoint(testEntryFilePath, 12));
+        addBreakPoint(new BallerinaTestDebugPoint(testEntryFilePath, 18));
+        addBreakPoint(new BallerinaTestDebugPoint(testEntryFilePath, 30));
+        addBreakPoint(new BallerinaTestDebugPoint(testEntryFilePath, 35));
+        addBreakPoint(new BallerinaTestDebugPoint(testEntryFilePath, 48));
+        initDebugSession(DebugUtils.DebuggeeExecutionKind.RUN);
+
+        // Test for debug engage in object init method
+        Pair<BallerinaTestDebugPoint, StoppedEventArguments> debugHitInfo = waitForDebugHit(10000);
+        Assert.assertEquals(debugHitInfo.getLeft(), testBreakpoints.get(0));
+
+        // Test for debug engage in object method
+        resumeProgram(debugHitInfo.getRight(), DebugResumeKind.NEXT_BREAKPOINT);
+        debugHitInfo = waitForDebugHit(10000);
+        Assert.assertEquals(debugHitInfo.getLeft(), testBreakpoints.get(1));
+
+        // Test for debug engage in remote object
+        resumeProgram(debugHitInfo.getRight(), DebugResumeKind.NEXT_BREAKPOINT);
+        debugHitInfo = waitForDebugHit(10000);
+        Assert.assertEquals(debugHitInfo.getLeft(), testBreakpoints.get(2));
+
+        // Test for debug engage in remote function
+        resumeProgram(debugHitInfo.getRight(), DebugResumeKind.NEXT_BREAKPOINT);
+        debugHitInfo = waitForDebugHit(10000);
+        Assert.assertEquals(debugHitInfo.getLeft(), testBreakpoints.get(3));
+
+        // Prepare variables for visibility test by adding a debug point at the end of the .bal file.
+        resumeProgram(debugHitInfo.getRight(), DebugResumeKind.NEXT_BREAKPOINT);
+        debugHitInfo = waitForDebugHit(10000);
+
+        Variable[] variables = fetchDebugHitVariables(debugHitInfo.getRight());
+        Arrays.sort(variables, compareByName);
+
+        // Variable visibility test for object method
+        assertVariable(variables[1], "v02_fullName", "John Doe", "string");
+
+        // Variable visibility test for remote object function
+        assertVariable(variables[3], "v04_statusCode", "500", "int");
```

Comment:
Is there any special reason for doing variable tests in here? We have separate tests for the variable validation right? Sorry If I've missed anything.

Reply:
This is to verify variable name, type and values are properly visible inside language constructs. Previous variable visibility tests do not cover this scenario.